fastlane/lib/fastlane/actions/gym.rb

```diff
@@ -12,11 +12,19 @@ def self.run(values)
         should_use_legacy_api = values[:use_legacy_build_api] || Gym::Xcode.pre_7?
 
         if values[:provisioning_profile_path].to_s.length.zero? && should_use_legacy_api
-          sigh_path = Actions.lane_context[Actions::SharedValues::SIGH_PROFILE_PATH] || ENV["SIGH_PROFILE_PATH"]
+          sigh_path = Actions.lane_context[SharedValues::SIGH_PROFILE_PATH] || ENV["SIGH_PROFILE_PATH"]
           values[:provisioning_profile_path] = File.expand_path(sigh_path) if sigh_path
         end
 
-        values[:export_method] ||= Actions.lane_context[Actions::SharedValues::SIGH_PROFILE_TYPE]
+        values[:export_method] ||= Actions.lane_context[SharedValues::SIGH_PROFILE_TYPE]
+
+        if Actions.lane_context[SharedValues::MATCH_PROVISIONING_PROFILE_MAPPING]
+          # Since Xcode 9 you need to provide the explicit provisioning profile per app target
```

Comment:
Since Xcode 9 you need to _explicitly provide_ the provisioning profile per app target